src/com/trilead/ssh2/Connection.java

```diff
@@ -666,6 +666,7 @@ final class TimeoutState
 			throw new IllegalArgumentException("kexTimeout must be non-negative!");
 
 		final TimeoutState state = new TimeoutState();
+		final TimeoutService timeoutService = new TimeoutService();		    
```

Comment:
now when there is a instance created for each connection couldn't be pass more information into  TimeoutService
like "new TimeoutService(hostname);"

which would help naming the threads
Trilead_TimeoutService_<hostname>_1

Reply:
@Elisedlund I like the suggestion since it will make traceability better. I will add hostname and hope @kuisathaverat  likes it too :-)